compiler-rt/lib/scudo/standalone/primary64.h

```diff
@@ -982,6 +999,21 @@ template <typename Config> class SizeClassAllocator64 {
     const uptr Size = getSizeByClassId(ClassId);
     const u16 MaxCount = CacheT::getMaxCached(Size);
 
+    MemMapT &RegionMemMap = Region->MemMapInfo.MemMap;
+    if (UNLIKELY(!RegionMemMap.isAllocated())) {
+      DCHECK_EQ(Region->RegionBeg, 0U);
+      ReservedMemoryT ReservedMemory;
+      ReservedMemory.create(/*Addr=*/0U, RegionSize, "scudo:primary_reserve");
```

Comment:
Does this only reserve the memory for this region?

Can it be modified to allow failure and return a nullptr? The person trying to use this sees a failure here, but it could mean that the allowed VA space has been exhausted.

Reply:
You're right. Let me do it

Reply:
Done. BTW, I did `push -f` to avoid having to cherry-pick all fix-ups.